corehq/apps/domain/forms.py

```diff
@@ -1147,6 +1181,28 @@ def clean(self):
                    "to send an email to the partner.").format(dimagi_user.username)
             self.add_error('dimagi_contact', msg)
 
+    def save_gsheet_limit(self, domain, limit_value, limit_type):
+        try:
+            limit = ProjectLimit.objects.get(domain=domain, limit_type=limit_type)
+
+            if self.cleaned_data.get('use_custom_gsheet_limit') != 'Y':
+                limit.delete()
+            else:
+                limit.limit_value = limit_value
+                limit.save()
+
+            return limit
+
+        except ProjectLimit.DoesNotExist:
+            if self.cleaned_data.get('use_custom_gsheet_limit') != 'Y':
+                return None
+            else:
+                return ProjectLimit.objects.create(
+                    domain=domain,
+                    limit_type=ProjectLimitType.LIVE_GOOGLE_SHEETS,
+                    limit_value=limit_value,
+                )
```

Comment:
It might be worth using [update_or_create](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#django.db.models.query.QuerySet.update_or_create) to simplify this object a bit. This method could probably be simplified into deleting the query if `use_custom_gsheet_limit` wasn't 'Y' (I believe deleting nothing doesn't result in errors), or using `update_or_create` otherwise.  This seems to only be called by the save code, which is hard-coded to always pass `ProjectLimitType.LIVE_GOOGLE_SHEETS` as the `limit_type`. Why can't we just remove `limit_type` and reference `ProjectLimitType.LIVE_GOOGLE_SHEETS` where it applies?